app/models/job.rb

```diff
@@ -91,7 +132,21 @@ def script_path
   end
 
   def allocation
-    FlightScheduler.app.allocations.for_job(self.id)
+    job_id = job_type == 'ARRAY_TASK' ? array_job.id : self.id
+    FlightScheduler.app.allocations.for_job(job_id)
+  end
+
+  def create_array_tasks
+    self.array_tasks ||= @array.split(',').map do |idx|
+      self.dup.tap do |task|
```

Comment:
This looks kinda strange to duplicate the job to generate the tasks. I believe we are conflating `task` and `job` into a single model when they are quite different objects.

Possible it would be better to create multiple `Task` decorators around a single `Job`. This way it would be easier for them to develop in different directions. `SimpleDelegator` would be a fairly easy first pass.

Bear in mind this is only a shallow duplication, so there maybe shared resource issue in the future

Reply:
You might be right about conflating a task object with a job object.  I expect that will become clearer as the remainder of array job/task functionality is implemented.

I've changed `create_array_tasks` to create new `Job`s instead of duplicating and adjusted the validations too.  I don't think you'll like the change any more than you do the existing mechanism, but it is much clearer in what attributes belong to which type of job.

Reply:
Looks like a step in the right direction. My gut instinct is eventually batch submissions would a `job`-`task` in a one-to-one relationship where array submissions are one-to-many.

However this is more LTE of the data model and can be shelved for now.